components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java

```diff
@@ -90,7 +95,325 @@ public class FIDOAuthenticator extends AbstractApplicationAuthenticator
+                initiateAuthenticationRequest(request, response, context);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+            } else {
+                if (enablePasskeyProgressiveEnrollment) {
+                    // If the user hasn't enrolled passkeys and if the passkey enrollment consent hasn't
+                    // received, then redirect the user to the consent page prior to initiating the passkey
+                    // enrollment request.
+                    if (!isPasskeyCreationConsentReceived(context)) {
+                        redirectToPasskeyEnrollmentConsentPage(response, context);
+                        return AuthenticatorFlowStatus.INCOMPLETE;
+                    }
+                    if (!FrameworkUtils.isPreviousIdPAuthenticationFlowHandler(context)) {
+                        return handlePasskeyEnrollmentScenarios(request, response, context);
+                    } else {
+                        persistUsername(context, authenticatedUser.getUserName());
+                        context.setProperty(IS_PASSKEY_CREATION_CONSENT_RECEIVED, true);
+                        return AuthenticatorFlowStatus.FAIL_COMPLETED;
+                    }
+                } else {
+                    // If passkeyProgressiveEnrollment is turned off, redirect users to the passkey status page to
+                    // inform them that they have no registered passkeys and can enroll through myAccount.
+                    redirectToPasskeysExistenceStatusPage(response, context, false);
+                    return AuthenticatorFlowStatus.INCOMPLETE;
+                }
+
+            }
+        } else {
+
+            if (enablePasskeyProgressiveEnrollment && isPasskeyCreationConsentReceived(context)) {
+                if (!StringUtils.isEmpty(request.getParameter(USER_NAME))) {
+                    persistUsername(context, request.getParameter(USER_NAME));
+                }
+                return AuthenticatorFlowStatus.FAIL_COMPLETED;
+            }
+
+            if (isFidoAsFirstFactor(context)) {
+
+                if (enableUsernamelessAuthentication) {
+                    initiateAuthenticationRequest(request, response, context);
+                    return AuthenticatorFlowStatus.INCOMPLETE;
+                }
+
+                if (StringUtils.isEmpty(request.getParameter(USER_NAME))) {
+                    redirectToFIDOIdentifierFirstPage(response, context);
+                    context.setProperty(FIDOAuthenticatorConstants.IS_IDF_INITIATED_FROM_AUTHENTICATOR, true);
+                    return AuthenticatorFlowStatus.INCOMPLETE;
+                }
+
+                // If an authentication request initiated from the custom FIDO identifier page, go through this flow.
+                if (!StringUtils.isEmpty(request.getParameter(SCENARIO)) &&
+                        request.getParameter(SCENARIO).equals(ScenarioTypes.INIT_FIDO_AUTH)) {
```

Comment:
Let's switch the constant to avoid any possible NPEs. Check other places as well

Reply:
Resolved via [59395d7](https://github.com/wso2-extensions/identity-local-auth-fido/pull/110/commits/59395d7a0c1dad27440c38d7ed768bb016868b68)